app/client/src/store/mutations.ts

```diff
@@ -19,4 +21,10 @@ export default {
   setIsolateValue(state: RootState, input: IsolateValue) {
     state.results.perIsolate[input.hash][input.type] = input.result;
   },
+  setStatus(state: RootState, status: Record<string, string>) {
+    state.analysisStatus[status.task as keyof AnalysisStatus] = status.data;
```

Comment:
is this `as` always ok? Is there a better way to do it than `as` here?

Reply:
I am not sure, maybe @EmmaLRussell can advise on a better solution?

Reply:
Could make the status an enum with string values, and use those both here and when defining the keys on analysisStatus